### PR TITLE
Require Jenkins 2.479 or newer

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Require Jenkins 2.479 or newer

JGit 7.0.0 requires Java 17 or newer.  Jenkins 2.475 requires Java 17 or newer and includes Jetty 12 and Jakarta EE 9.  Jenkins 2.479 includes a security fix and is the baseline for the next LTS.

Require the Jenkins weekly is the baseline for the next LTS.

Would be nice to wait on this release until the plugin BOM is available for 2.479.x so that it can depend on the 2.479.x plugin BOM.  That should happen in the plugin BOM release 15 Oct 2024.

### Testing done

Interactive testing looks good and automated tests all pass.

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by basil** [comment]
Should also migrate to EE 9:

```java
src/main/java/hudson/plugins/git/GitTool.java
30:import org.kohsuke.stapler.StaplerRequest;
156:        public boolean configure(StaplerRequest req, JSONObject json) {
```

**Review by basil — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -62,10 +62,8 @@
     <jenkins.baseline>2.462</jenkins.baseline>
     <!-- TODO Replace with the standard jenkins.baseline references after LTS requires Java 17  -->
     <!-- <jenkins.version>${jenkins.baseline}.1</jenkins.version> -->
```
Comment (jglick) [inline pom.xml:64]: https://github.com/jenkinsci/bom/pull/3677 IIUC

Comment (MarkEWaite) [reply pom.xml:64]: BOM will release Friday, then I plan to release the git client plugin 6.1.0 using that new plugin BOM release.

Comment (MarkEWaite) [reply pom.xml:64]: Or maybe it is better to release git client plugin 6.1.0 before the weekly BOM so that it is available sooner to BOM consumers

Comment (MarkEWaite) [reply pom.xml:64]: The more I think about it, the more I think it is better to release this today then include it in the plugin BOM release tomorrow

**Pull request merged**